apps/prairielearn/elements/pl-string-input/pl-string-input.mustache

```diff
@@ -1,66 +1,125 @@
+<script>
+    $(function(){
+        document.querySelectorAll('.multiline-input#pl-string-input-{{uuid}}-area').forEach((input) => {
+            input.addEventListener('input', function () {
+            // Adjusts the height based on the feedback content. If the feedback changes, the height
+            // changes as well. This is done by resetting the height (so the scrollHeight is computed
+            // based on the minimum height) and then using the scrollHeight plus padding as the new height.
+            this.style.height = '';
+            if (this.scrollHeight) {
+                const style = window.getComputedStyle(this);
+                this.style.height =
+                this.scrollHeight + parseFloat(style.paddingTop) + parseFloat(style.paddingBottom) + 'px';
+            }
+            });
+            input.dispatchEvent(new Event('input'));
+        });
+    });
+</script>
+
 {{#question}}
+
 {{#inline}}<span class="form-inline d-inline-block ml-2">{{/inline}}
-    <span class="input-group pl-string-input">
+    {{#multiline}}
         {{#label}}
-        <span class="input-group-prepend">
-            <span class="input-group-text" id="pl-string-input-{{uuid}}-label">{{{label}}}</span>
-        </span>
+        <label
+            id="pl-string-input-{{uuid}}-label"
+            class="form-label"
+            for="pl-string-input-{{uuid}}-area"
+        >{{{label}}}</label>
         {{/label}}
-        <input
-            name="{{name}}"
-            type="text"
-            inputmode="text"
-            class="form-control pl-string-input-input"
-            size="{{size}}"
-            autocomplete="off"
-            autocorrect="off"
-            {{^editable}}disabled{{/editable}}
-            {{#raw_submitted_answer}}value="{{raw_submitted_answer}}"{{/raw_submitted_answer}}
-            aria-describedby="pl-symbolic-input-{{uuid}}-suffix"
-            aria-labelledby="pl-symbolic-input-{{uuid}}-label"
-            placeholder="{{placeholder}}"
-        />
-        <span class="input-group-append">
-            {{#suffix}}<span class="input-group-text" id="pl-symbolic-input-{{uuid}}-suffix">{{suffix}}</span>{{/suffix}}
-
-            {{#show_info}}
-                <a role="button" class="btn btn-light border d-flex align-items-center" data-toggle="popover" data-html="true" title="String" data-content="{{info}}" data-placement="auto" data-trigger="focus" tabindex="0">
-                    <i class="fa fa-question-circle" aria-hidden="true"></i>
-                </a>
-            {{/show_info}}
+        <span id="pl-string-input-{{uuid}}" class="input-group pl-string-input">
+            <textarea
+                wrap="soft"
+                id="pl-string-input-{{uuid}}-area"
+                name="{{name}}"
+                type="text"
+                inputmode="text"
+                class="form-control pl-string-input-input multiline-input"
```

Comment:
I'd strongly encourage you to use an element-specific class here, like `pl-string-input-multiline`. This is a best practice to avoid collisions with other things on the page.

Reply:
https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/034e1035f27db64dbf47102accab254a7e002645